test-e2e/fixtures/zk_scripts/dcos-zk-backup.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+SSH_USER='root'
+REMOTE_TMP_DIR=/tmp
+
+usage() {
+  echo "Usage: $0 [-h] [-l SSH_USER] [-i SSH_KEY_PATH] [-t REMOTE_TMP_DIR] ZK_BACKUP_DEST_DIR MASTER_HOSTNAME" 1>&2
+}
+exit_abnormal() {
+  usage
+  exit 1
+}
+while getopts ":hl:i:t:" opt; do
+  case $opt in
+    h)
+      usage
+      exit 0
+    ;;
+    l) SSH_USER=$OPTARG
+    ;;
+    i) SSH_KEY_PATH=$OPTARG
+    ;;
+    t) REMOTE_TMP_DIR=$OPTARG
+    ;;
+    \?) echo "Error: Invalid option -$OPTARG" 1>&2
+        exit_abnormal
+    ;;
+    :) echo "Error: -$OPTARG requires an argument." 1>&2
+       exit_abnormal
+    ;;
+    *) exit_abnormal
+    ;;
+  esac
+done
+
+shift $((OPTIND - 1))
+if (($# != 2)); then
+  exit_abnormal
+fi
+
+DESTINATION_DIR=$1
+if [ ! -d "$DESTINATION_DIR" ]; then
+  echo "Error: $DESTINATION_DIR does not exist." 1>&2
+  exit 1
+fi
+
+MASTER_HOST=$2
+REMOTE_BACKUP_DIR=${REMOTE_TMP_DIR}/zk_backup.$$
+KEY=${SSH_KEY_PATH:+"-i $SSH_KEY_PATH"}
```

Comment:
As per our discussion (but note for others): We cannot guarantee that every master will use the same SSH key.